src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ISizeOfExpression.cs

```diff
@@ -0,0 +1,228 @@
+  Instance Receiver: ISizeOfExpression (OperationKind.SizeOfExpression, Type: System.Int32, IsInvalid) (Syntax: 'sizeof(M2')
+      TypeOperand: M2
+  Arguments(0)
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS1026: ) expected
+                //         i = /*<bind>*/sizeof(M2()/*</bind>*/);
+                Diagnostic(ErrorCode.ERR_CloseParenExpected, "(").WithLocation(8, 32),
+                // CS1002: ; expected
+                //         i = /*<bind>*/sizeof(M2()/*</bind>*/);
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, ")").WithLocation(8, 45),
+                // CS1513: } expected
+                //         i = /*<bind>*/sizeof(M2()/*</bind>*/);
+                Diagnostic(ErrorCode.ERR_RbraceExpected, ")").WithLocation(8, 45),
+                // CS0246: The type or namespace name 'M2' could not be found (are you missing a using directive or an assembly reference?)
+                //         i = /*<bind>*/sizeof(M2()/*</bind>*/);
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "M2").WithArguments("M2").WithLocation(8, 30),
+                // CS0233: 'M2' does not have a predefined size, therefore sizeof can only be used in an unsafe context (consider using System.Runtime.InteropServices.Marshal.SizeOf)
+                //         i = /*<bind>*/sizeof(M2()/*</bind>*/);
+                Diagnostic(ErrorCode.ERR_SizeofUnsafe, "sizeof(M2").WithArguments("M2").WithLocation(8, 23)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<InvocationExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void TestSizeOfExpression_MissingArgument()
+        {
+            string source = @"
+using System;
+
+class C
+{
+    void M(int i)
+    {
+        i = /*<bind>*/sizeof()/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+ISizeOfExpression (OperationKind.SizeOfExpression, Type: System.Int32, IsInvalid) (Syntax: 'sizeof()')
+  TypeOperand: ?
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS1031: Type expected
+                //         i = /*<bind>*/sizeof()/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_TypeExpected, ")").WithLocation(8, 30),
+                // CS0233: '?' does not have a predefined size, therefore sizeof can only be used in an unsafe context (consider using System.Runtime.InteropServices.Marshal.SizeOf)
+                //         i = /*<bind>*/sizeof()/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_SizeofUnsafe, "sizeof()").WithArguments("?").WithLocation(8, 23)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<SizeOfExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+    }
+}
```

Comment:
There is no VB equivalent of `sizeof` to test, correct?

Reply:
Yes, I could not find any.